packages/frontend/src/components/dialogs/ProxyConfiguration/index.tsx

```diff
@@ -0,0 +1,380 @@
+          proxy_enabled: proxyState.enabled ? Proxy.ENABLED : Proxy.DISABLED,
+        })
+
+        await BackendRemote.rpc.stopIo(accountId)
+        await BackendRemote.rpc.startIo(accountId)
+      } catch (error: any) {
+        log.error('failed to update proxy settings', error)
+        openAlertDialog({
+          message: `${tx('error')} ${error.message || error.toString()}`,
+        })
+      }
+    }
+    updateProxySettings()
+  }, [
+    proxyState.enabled,
+    accountId,
+    openAlertDialog,
+    tx,
+    proxyState.proxies,
+    proxyState.activeProxy,
+    proxyState.updateSettings,
+    proxyState,
+  ])
+
+  /**
+   * validate settings before closing
+   * to avoid invalid settings
+   */
+  const closeDialog = () => {
+    if (proxyState.enabled && proxyState.proxies.length === 0) {
+      openAlertDialog({
+        message: tx('proxy_invalid'),
+      })
+      return
+    }
+    onClose()
+  }
+
+  const copyToClipboard = useCallback(
+    (url: string) => {
+      navigator.clipboard.writeText(url)
+      openAlertDialog({
+        message: `${url}\n${tx('copied_to_clipboard')}`,
+      })
+    },
+    [tx, openAlertDialog]
+  )
+
+  return (
+    <Dialog
+      fixed
+      width={400}
+      dataTestid='proxy-dialog'
+      canOutsideClickClose={false}
+    >
+      <DialogHeader
+        title={tx('menu_settings')}
+        onClose={closeDialog}
+        dataTestid='proxy-settings'
+      />
+      <DialogBody className={styles.proxyDialogBody}>
+        <div className={styles.container}>
+          <SettingsSwitch
+            label={tx('proxy_use_proxy')}
+            value={proxyState.enabled}
+            onChange={changeProxyEnable}
+          />
+          <div>
+            <h3 className='title'>{tx('proxy_list_header')}</h3>
+            <p className={styles.explain}>{tx('proxy_add_explain')}</p>
+          </div>
+          <div className={styles.proxyList} role='radiogroup'>
+            {proxyState.proxies.map((proxyUrl, index) => (
+              <ProxyItemRow
+                key={index}
+                proxyUrl={proxyUrl}
+                index={index}
+                isActive={proxyUrl === proxyState.activeProxy}
+                configured={configured}
+                changeActiveProxy={changeActiveProxy}
+                connectivityStatus={connectivityStatus}
+                copyToClipboard={copyToClipboard}
+                deleteProxy={deleteProxy}
+              />
+            ))}
+          </div>
+        </div>
+      </DialogBody>
+      <DialogFooter>
+        {showNewProxyForm && (
+          <>
+            <DeltaInput
+              label={tx('proxy_add_url_hint')}
+              value={newProxyUrl}
+              onChange={e => setNewProxyUrl(e.target.value)}
+            />
+            <Button
+              className='save-proxy'
+              onClick={() => addProxy(newProxyUrl)}
+            >
+              {tx('proxy_add')}
+            </Button>
+          </>
```

Comment:
It would be nice to make this an actual `<form>` so that you can submit with "Enter". But this is fine I guess.

Reply:
I changed it by just using a form tag which has no additional attributes. Not sure what the impact on accessibility is here...

Reply:
Yes, now "Enter" works. Although it'd be nice to add `type="submit"` on the button for good measure.